framework/source/class/qx/data/controller/ListOfObjects.js

```diff
@@ -0,0 +1,444 @@
+      PROPERTIES
+   *****************************************************************************
+   */
+
+  properties :
+  {
+    /** Data array containing the data which should be shown in the list. */
+    model :
+    {
+      check: "qx.data.IListData",
+      apply: "_applyModel",
+      event: "changeModel",
+      nullable: true,
+      dereference: true
+    },
+
+
+    /** The target widget which should show the data. */
+    target :
+    {
+      apply: "_applyTarget",
+      event: "changeTarget",
+      nullable: true,
+      init: null,
+      dereference: true
+    },
+
+
+    /**
+     * The path to the property which holds the information that should be
+     * shown as a label. This is only needed if objects are stored in the model.
+     */
+    labelPath :
+    {
+      check: "String",
+      apply: "_applyLabelPath",
+      nullable: true
+    },
+
+
+    /**
+     * The path to the property which holds the information that should be
+     * shown as an icon. This is only needed if objects are stored in the model
+     * and if the icon should be shown.
+     */
+    iconPath :
+    {
+      check: "String",
+      apply: "_applyIconPath",
+      nullable: true
+    },
+
+
+    /**
+     * A map containing the options for the label binding. The possible keys
+     * can be found in the {@link qx.data.SingleValueBinding} documentation.
+     */
+    labelOptions :
+    {
+      apply: "_applyLabelOptions",
+      nullable: true
+    },
+
+
+    /**
+     * A map containing the options for the icon binding. The possible keys
+     * can be found in the {@link qx.data.SingleValueBinding} documentation.
+     */
+    iconOptions :
+    {
+      apply: "_applyIconOptions",
+      nullable: true
+    },
+
+
+    /**
+     * Delegation object, which can have one or more functions defined by the
+     * {@link qx.data.controller.IArrayDelegate} interface.
+     */
+    delegate :
+    {
+      apply: "_applyDelegate",
+      event: "changeDelegate",
+      init: null,
+      nullable: true
+    }
+  },
+
+
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+  members :
+  {
+    
+    
+    /*
+    ---------------------------------------------------------------------------
+       PUBLIC API
+    ---------------------------------------------------------------------------
+    */
+    reload: function() {
+      this.__items.reload();
+    },
+
+    
+    
+    /*
+    ---------------------------------------------------------------------------
+       ITEMS ARRAYMIRROR DELEGATE METHODS
+    ---------------------------------------------------------------------------
+    */
+    _onCreateTargetItem: function() {
+      var delegate = this.getDelegate();
+      if (delegate && typeof delegate.createItem == "function")
```

Comment:
referring to IArrayDelegate I guess this must be delegate.createTargetItem, right?